protocol/database-scripting/README.md

```diff
@@ -0,0 +1,102 @@
+Database Scripting
+==============
+
+A protocol for running console scripts on the Healthify app's production database.
+
+Development
+-----------
+
+Develop the scripting story similarly to any other story. In this case, this would
+entail running your script at the rails console of your local server.
+
+While some scripts only lend themselves to being performed via pure SQL, where it
+is possible, you should leverage ActiveRecord to perform our INSERT/UPDATE/DELETE
+operations so that we leverage ActiveRecord validations and callbacks.
+
+While developing the script, it is critical to plan our remediation strategy if
+something does go awry.
+
+One useful approach to consider is to wrap the INSERT/UPDATE/DELETE commands within
+a transaction, and then to assert on certain states of the database after the script
+is complete, rolling back all changes if the assertions fail.
+
+Further, it's important to be able to execute a complete reversal of your changes
+in the event that there are unexpected failures. Ideally, we would write a reversal
+script to immediately reverse our changes if necessary.
+
+For INSERT operations, we would need to retain data about what records ended up
+being INSERT-ed by our script, and then the reversal script would delete those
+records.
+
+For UPDATE/DELETE operations, we would want to dump the data of all the records
+we will be interacting with into a Temp Table (or Tables). Our reversal script would then
+perform a COPY of the data from the Temp Table into the original table (and a deletion
+of updated records in the case of an Update).
+
+Code Review
+-----------
+
+Typically, console scripts are not code that we want to add to the Healthify app's
+code repo. As a result, to solicit Code Review you can put your code in a Github
+[gist](https://help.github.com/articles/about-gists/) and share the gist's link
+with your teammates to get Code Review. Also, be sure to include a link to the gist
+in the description of the corresponding Pivotal Story.
+
+Running database scripts poses a higher risk than deploying a normal feature. Hence,
+it's important to have a higher bar of Code Review than for normal stories. Make sure
+you solicit Code Review from at least two independent reviewers.
+
+Pre-execution + QA
+-------------------
+
+Before executing the script, generate a backup of the database you will be running the script
+on. This can be done with the Aptible CLI. E.g.
+
+```
+  aptible db:backup production-healthify-postgresql
+```
+
+You can view database backups in the Aptible Dashboard by viewing the show page of
+a particular database and clicking the "Backups" tab. Take note of which backup
+you just created (the backups are timestamped). In the event that there is a problem,
+the backup can be used as the source of truth about what the affected database *should*
+look like.
```

Comment:
Do you think it would be important to note that we should be using back-ups from `us-east-1`, instead of `us-west-1`? This is something that we ran into while debugging with Frank.